docs/rfcs/xxx-pretranspile.md

```diff
@@ -0,0 +1,44 @@
+# Pre-transpiled Atom packages
+
+## Status
+
+Proposed
+
+## Summary
+
+This feature will enable package authors to use conventional npm tooling and package.json conventions to take advantage of JavaScript transpilers like Babel or TypeScript.
+
+## Motivation
+
+Transpiling packages on _publish_ rather than _load_ will have great benefits for package authors:
```

Comment:
does transpiling on publish, rather than on load, also have the benefit of making test coverage simpler (for package authors who want to set that up?)

Reply:
Not necessarily. I haven't added anything to this about packages running in dev mode -- for the moment, dev mode, including the test runner, will still use Atom's transpiler pipeline. Although you could set up your `package.json` to `prepare` before running tests... ?

Istanbul also has a problem with Electron subprocesses not inheriting the code it's trying to inject properly which this won't address.